OffchainLabs/arbitrum-token-bridge
# fix: prevent unnecessary refetch for withdrawals

### Summary

There were a couple of things that were causing unnecessary re-renders:
* `useL2Gateways`
  * Since it had `l2Provider` as a prop, it would only run the effect to update after the provider changes, which would be one render late
* `walletAddress` being `undefined`
  * Even though the types doesn't state it, 
  * I've only added comments explaining it, since it will be removed in https://github.com/OffchainLabs/arbitrum-token-bridge/pull/938
* `pageSize`
  * Similar to `useL2Gateways`, the prop would lag behind a render, so instead I memoized it based on the network, and overrode the default `pageSize`
  
All these lagging changes would cause the SWR query keys to update multiple times and start multiple requests.

### Steps to test

It's kinda hard to test without adding console.logs all over the app and observing how SWR query keys change, but one thing that you can do is:
* Connect to Arbitrum Sepolia, and have Sepolia as the destination
* Open up the Network tab and clear out requests
* Switch the destination to Stylus Testnet
* See the request count, let's mark it as A
* Then refresh the page
* See the request count, let's mark it as B
* A & B should be roughly the same, whereas if you try it out bridge.arbitrum.io, A would be roughly 2 times B (it's easier to notice on accounts that have a lot of withdrawals)
